sanitiser/_ids.js

```diff
@@ -34,16 +34,20 @@ function sanitizeId(rawId, messages) {
     messages.errors.push( formatError(rawId) );
     return;
   }
-
-  if (!_.includes(type_mapping.sources, source)) {
-    messages.errors.push( targetError(source, type_mapping.sources) );
+  var valid_values = Object.keys(type_mapping.source_mapping);
+  if (!_.includes(valid_values, source)) {
+    messages.errors.push( targetError(source, valid_values) );
     return;
   }
 
   if (!_.includes(type_mapping.layers, layer)) {
     messages.errors.push( targetError(layer, type_mapping.layers) );
     return;
   }
+  //converts the shortened source names to the full name
```

Comment:
You could remove the if statement here, since the source_mapping supports either the short or long name